parachain/pallets/verifier-lightclient/src/lib.rs

```diff
@@ -331,6 +334,13 @@ impl<T: Config> Module<T> {
 			)?;
 			if new_finalized_block_id != finalized_block_id {
 				FinalizedBlock::put(new_finalized_block_id);
+				Headers::<T>::mutate(new_finalized_block_id.hash, |option| -> DispatchResult {
+					if let Some(header) = option {
+						header.finalized = true;
+						return Ok(());
+					}
+					Err(Error::<T>::Unknown.into())
+				})?;
```

Comment:
Do we also have to iterate through previous headers to mark them as finalized, up until the highest previous finalized header?

Reply:
No. This finalized header's parent would have been marked as finalized when `header.number - 1` was imported, and so forth for each ancestor